### ISSUE TITLE
Docker Setup Quality of Life Update

### ISSUE THREAD

**Issue by Jquinny** [UAlbertaFSAE/f1tenth#73]
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is.

Our docker setup works right now, but it's slightly inefficient in how the dependencies are downloaded into each container. The containers are also not as self contained as environment variables are not all baked into them. There is also a lot to keep track of in terms of dockerfiles and compose configurations. The DISPLAY env variable is also kind of funky right now with WSL and ssh environments, getting that figured out would be nice.

**Describe the solution you'd like**

Since we use an ubuntu 22.04 container for development and for jetson, we should be able to make a combined base image that adds the user (we don't need to use the user on the jetson but it can be there anyhow) and base dependencies/tooling, base environment variables, etc. Then, we can build separate development and jetson containers from that, which contain specific things for each of them like the zed stuff in the jetson container and the sim stuff in the development one.

We could look at combining the sim stuff into the dev container and the hardware stuff into the jetson container, decreasing the amount of dockerfile and container configurations we need.

**Note:** This is definitely not a priority as things work fine for now, but it would be a quality of life update and potentially make builds more efficient and less large.

TL:DR things that could be done:
- combine base setup and dependencies into a base image, build dev and jetson containers off of base container
- make containers more self contained by adding all necessary environment variables into them
- optimize image configurations for speed and resource usage
- DISPLAY environment variable maps correctly depending on device configuration

**Describe alternatives you've considered**

**Additional context**
